cmd/proxy/main.go

```diff
@@ -0,0 +1,173 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/handler"
+	"k8s.io/kube-openapi/pkg/handler3"
+	"k8s.io/kube-openapi/pkg/spec3"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+)
+
+func serve_v2(swagger spec.Swagger, h common.PathHandler) error {
+	o, err := handler.NewOpenAPIService(&swagger)
+	if err != nil {
+		return fmt.Errorf("failed to create OpenAPI v2 service: %v", err)
+	}
+	err = o.RegisterOpenAPIVersionedService("/openapi/v2", h)
+	if err != nil {
+		return fmt.Errorf("failed to register OpenAPI service: %v", err)
+	}
+	return nil
+}
+
+func serve_v3(v3 map[XKubernetesGroupVersion]spec3.OpenAPI, h common.PathHandler) error {
+	o, err := handler3.NewOpenAPIService(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create OpenAPI v3 service: %v", err)
+	}
+
+	h.Handle("/openapi/v3", http.HandlerFunc(o.HandleDiscovery))
+	for gv := range v3 {
+		h.Handle("/openapi/v3/"+gv.Path(), http.HandlerFunc(o.HandleGroupVersion))
+		spec := v3[gv]
+		o.UpdateGroupVersion(gv.Path(), &spec)
+	}
+	return nil
+}
+
+func parse_v2(swagger_json []byte) (*spec.Swagger, error) {
+	swagger := spec.Swagger{}
+	if err := json.Unmarshal(swagger_json, &swagger); err != nil {
+		return nil, fmt.Errorf("failed to parse json: %v", err)
+	}
+	return &swagger, nil
+}
+
+type XKubernetesGroupVersion struct {
```

Comment:
Does this type exist because we cant import k8s types directly from kube-openapi?

Reply:
I don't think anything currently parses the XKubernetesGroupVersion directly from the openapi extension? What type did you have in mind?

Reply:
Not really sure how this type does anything special to parse the openapi exnteion. It seems like a copy of https://github.com/kubernetes/kubernetes/blob/a9593d634c6a053848413e600dadbf974627515f/staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L166-L183 except with a `Path()` method.

but you wouldn't be able to use that due to import cycle

Reply:
Yeah, it's mostly the same thing, it's also only a few lines of code, whatever?

Reply:
> Does this type exist because we cant import k8s types directly from kube-openapi?

Is this a bigger issue that we should look to resolve? That's the same reason we can't use the HandlePrefix in this repo.